lib/default.nix

```diff
@@ -260,17 +257,19 @@ in {
     && stdenv.buildPlatform == stdenv.hostPlatform
     && stdenv.hostPlatform == stdenv.targetPlatform;
 
-  # Takes a version number or attr set of arguments (for cabalProject)
-  # and converts it to an attr set of arguments.  This allows
+  # Takes a version number, module or list of modules (for cabalProject)
+  # and converts it to an list of project modules.  This allows
   # the use of "1.0.0.0" or { version = "1.0.0.0"; ... }
-  versionOrArgsToArgs = versionOrArgs:
-    if lib.isAttrs versionOrArgs
-      then versionOrArgs
-      else { version = versionOrArgs; };
+  versionOrModToMods = versionOrMod:
+    if lib.isString versionOrMod
+      then [{ version = versionOrMod; }]
+    else if lib.isList versionOrMod
+      then versionOrMod
+    else [versionOrMod];
 
   # Find the resolver in the stack.yaml file and fetch it if a sha256 value is provided
   fetchResolver = import ./fetch-resolver.nix {
-    inherit (pkgs.evalPackages) pkgs;
+    inherit (pkgs.buildPackages) pkgs;
```

Comment:
similarly

Reply:
`fetchResolver` is only called from `lib/stack-cache-generator.nix` and that only has access to `evalPackages` now (does not have `pkgs` at all).